### ISSUE TITLE
JDK 9 java.lang.NoClassDefFoundError: javax/annotation/Priority

### ISSUE THREAD

**Issue by Nakaner** [graphhopper/graphhopper#1391]
Both GraphHopper and the map matching project can be build with JDK 9 but running the web API fails:

```
michael@thishost:~/graphhopper$ /usr/lib/jvm/java-9-openjdk-amd64/bin/java -Xmx6000m -Xms1000m -Dprepare.ch.weightings=no -Dgraphhopper.datareader.file=../map-matching/map-data/leipzig_germany.osm.pbf -Dgraph.location=graph-cache -jar web/target/graphhopper-web-0.11-SNAPSHOT.jar serve config-example.yml 
java.lang.NoClassDefFoundError: javax/annotation/Priority
	at org.glassfish.jersey.model.internal.ComponentBag.modelFor(ComponentBag.java:483)
	at org.glassfish.jersey.model.internal.ComponentBag.access$100(ComponentBag.java:89)
	at org.glassfish.jersey.model.internal.ComponentBag$5.call(ComponentBag.java:408)
	at org.glassfish.jersey.model.internal.ComponentBag$5.call(ComponentBag.java:398)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:228)
	at org.glassfish.jersey.model.internal.ComponentBag.registerModel(ComponentBag.java:398)
	at org.glassfish.jersey.model.internal.ComponentBag.register(ComponentBag.java:309)
	at org.glassfish.jersey.model.internal.CommonConfig.register(CommonConfig.java:466)
	at org.glassfish.jersey.server.ResourceConfig.register(ResourceConfig.java:453)
	at io.dropwizard.jersey.DropwizardResourceConfig.<init>(DropwizardResourceConfig.java:67)
	at io.dropwizard.jersey.DropwizardResourceConfig.<init>(DropwizardResourceConfig.java:47)
	at io.dropwizard.setup.Environment.<init>(Environment.java:80)
	at io.dropwizard.cli.EnvironmentCommand.run(EnvironmentCommand.java:37)
	at io.dropwizard.cli.ConfiguredCommand.run(ConfiguredCommand.java:87)
	at io.dropwizard.cli.Cli.run(Cli.java:78)
	at io.dropwizard.Application.run(Application.java:93)
	at com.graphhopper.http.GraphHopperApplication.main(GraphHopperApplication.java:46)
Caused by: java.lang.ClassNotFoundException: javax.annotation.Priority
	at jdk.internal.loader.BuiltinClassLoader.loadClass(java.base@9-internal/BuiltinClassLoader.java:366)
	at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(java.base@9-internal/ClassLoaders.java:184)
	at java.lang.ClassLoader.loadClass(java.base@9-internal/ClassLoader.java:419)
	... 19 more
```

The steps before:

```sh
# Git repository cloned
mvn package -DskipTests
```

It doesn't matter if I use the `graphhopper.sh` script or not (the script chooses Maven at /usr/share/maven)

Java and Maven version:

```
michael@thishost:~/map-matching$ java -version
openjdk version "9-internal"
OpenJDK Runtime Environment (build 9-internal+0-2016-04-14-195246.buildd.src)
OpenJDK 64-Bit Server VM (build 9-internal+0-2016-04-14-195246.buildd.src, mixed mode)
michael@thishost:~/map-matching$ mvn -version
Apache Maven 3.3.9
Maven home: /usr/share/maven
Java version: 9-internal, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-9-openjdk-amd64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.14.12-xxxx-std-ipv6-64", arch: "amd64", family: "unix"
```

I observed the issue with my own routing and map matching project today when I was trying to deploy it on a machine with JDK 9. This issue is somehow related to the new module system which is new with JDK 9.

**Comment by mprins** [comment]
seems similar to #1388

**Comment by Nakaner** [comment]
I have the impression that the JDK 9 which is available on Ubuntu 16.04 is not the JDK 9 I expected. It does not support the `--add-module` option although it should. However, I could compile my own routing project which had the same issue, successfully on a Ubuntu 18.04 machine with JDK 10.
